src/model/openthaigpt_pretraining_model/lightning/utils.py

```diff
@@ -223,13 +223,18 @@ def val_step(self):
     def train(self):
         self.opt.zero_grad()
         for epoch in range(self.start_epochs, self.epochs):
+            if epoch < self.start_epochs:
```

Comment:
I think `epoch` has been skipped by `for epoch in range(self.start_epochs, self.epochs)`.

Reply:
Oh, I just saw.

Reply:
Now,I already remove it.